tracing.js

```diff
@@ -0,0 +1,58 @@
+"use strict";
+
+const tracing = require("signalfx-tracing");
+const opentracing = require("opentracing");
+const NoopTracer = require("signalfx-tracing/src/noop/tracer");
+
+// must init tracer here before other libraries are imported in order to apply
+// auto-instrumentation patches.
+let _tracer;
+
+if ((process.env.SIGNALFX_TRACING_ENABLED || "true").toLowerCase() == "false") {
+  _tracer = new NoopTracer();
+} else {
+  const options = {
+    service:
+      process.env.SIGNALFX_SERVICE_NAME || process.env.AWS_LAMBDA_FUNCTION_NAME,
+  };
+  if (process.env.SIGNALFX_ENDPOINT_URL) {
+    options.url = process.env.SIGNALFX_ENDPOINT_URL;
+  }
+  _tracer = tracing.init(options);
+}
+
+opentracing.initGlobalTracer(_tracer);
+
+module.exports = {
+  tracer: function (tracingDisabled) {
+    return tracingDisabled ? new NoopTracer() : _tracer;
+  },
+
+  flush: function () {
+    return _tracer.flush();
+  },
+
+  startSpan: function (tracer, event, meta) {
+    const opName = "lambda_node_" + meta.aws_function_name;
+    let childOf = tracer.extract(
+      opentracing.FORMAT_HTTP_HEADERS,
+      event.headers || {}
+    );
+
+    var span;
+    if (childOf) {
+      span = tracer.startSpan(opName, { childOf });
+    } else {
+      span = tracer.startSpan(opName);
+    }
+
+    Object.entries(meta).forEach((pair) => {
+      if (pair[1]) {
+        span.setTag(pair[0], pair[1]);
+      }
+    });
+    span.setTag("span.kind", "server");
+    span.setTag("component", "node-lambda-wrapper");
```

Comment:
Wouldn't be better to have these 2 before filling up the ones from `meta`? This way they can be overwritten.

Reply:
Not sure. Should they be overwritten? :) I think they should pretty much always be present when this tracer is used. May be user can manually override them but don't think we should add a chance of having random metadata override these tags.

Reply:
I was understanding the "metadata" as explicit configured by the user if this is "incoming" metadata from the request then better as it is.

Reply:
Yes, this is mostly extracted from lambda context i.e name of lambda function, version, runtime etc.